cdci_data_analysis/analysis/parameters.py

```diff
@@ -674,29 +679,27 @@ def _set_angle(self, value, units):
 #
 
 
+class SpectralBoundary(Float):
+    def __init__(self, value=None, E_units='keV', name=None):
 
-class SpectralBoundary(Parameter):
-    def __init__(self,value=None,E_units='keV',name=None):
-
-        _allowed_units = ['keV','eV','MeV','GeV','TeV','Hz','MHz','GHz']
-
+        _allowed_units = ['keV', 'eV', 'MeV', 'GeV', 'TeV', 'Hz', 'MHz', 'GHz']
+        # TODO is this a reasonable assumption for this Psrameter? And for the others too?
+        if E_units is None or E_units == '':
+            E_units = 'keV'
         #wtform_dict = {'keV': FloatField}
 
-        super(SpectralBoundary, self).__init__(value=value,
-                                   units=E_units,
-                                   check_value=self.check_energy_value,
-                                   name=name,
-                                   allowed_units=_allowed_units)
-                                   #wtform_dict=wtform_dict)
-
-
-
+        super().__init__(value=value,
+                         units=E_units,
+                         default_units='keV',
+                         check_value=self.check_energy_value,
+                         name=name,
+                         allowed_units=_allowed_units)
+                           #wtform_dict=wtform_dict)
 
     @staticmethod
     def check_energy_value(value, units=None,name=None):
         #print('check type of ',name,'value', value, 'type',type(value))
```

Comment:
this method body is repeated multiple times. Can it be just kept in `Float` (with optimizations)?

Reply:
it will be there, it makes sense